lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -2156,7 +2631,9 @@ class MongoClientInterface {
                             return next(errors.InternalError);
                         }
                         bucketCount++;
-                        bucketInfos.push(bucketInfo);
+                        if (bucketInfo) {
```

Comment:
not possible, value is passed unless errors... We could try to type the callback like this:

```
interface ArsenalCallback<T> {
    (err: null, result: T): void;
    (err: ArsenalError, result: undefined): void;
};
```

...but while it may ensure the callback is used (in the function) as expected, typescript may not be smart enough to understand that, or even stop type inferrence because of the overloads.

seems like a good use for null-assertion though...
```suggestion
                        bucketInfos!.push(bucketInfo);
```